src/vm/classcompat.cpp

```diff
@@ -2596,9 +2596,14 @@ VOID    MethodTableBuilder::EnumerateClassMethods()
             }
         }
 
-#ifndef FEATURE_DEFAULT_INTERFACES
         // Some interface checks.
-        if (fIsClassInterface)
+        // Ngen wasn't upgraded to deal with default interface methods.
+        if (fIsClassInterface
+            && IsCompilationProcess()
+#if FEATURE_READYTORUN
```

Comment:
You do not need this ifdef. `IsReadyToRunCompilation` is implemented to return false when `FEATURE_READYTORUN` is not compiled.